Tools/CISupport/ews-build/steps.py

```diff
@@ -6874,3 +6905,330 @@ def doStepIf(self, step):
+        num_issues = self.getProperty('unexpected_new_issues', 0)
+        failing_files = (", ").join(self.getProperty('failures', []))
+        passing_files = (", ").join(self.getProperty('passes', []))
+        results_summary = ''
+
+        results_link = self.resultDirectoryURL() + SCAN_BUILD_OUTPUT_DIR + "/new-results.html"
+        build_link = f'{self.master.config.buildbotURL}#/builders/{self.build._builderid}/builds/{self.build.number}'
+        formatted_build_link = f'[#{self.getProperty("buildnumber", "")}]({build_link})'
+
+        if num_failures:
+            pluralSuffix = 's' if num_issues > 1 else ''
+            comment = f"Smart Pointer Build {formatted_build_link}: Found [{num_issues} new failure{pluralSuffix}]({results_link}), blocking PR #{self.getProperty('github.number')}.\n"
+            comment += 'Please address these issues before landing. See [WebKit Guidelines for Safer C++ Programming](https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines).\n(cc @rniwa)'
+            self.setProperty('comment_text', comment)
+            self.build.addStepsAfterCurrentStep([LeaveComment(), SetCommitQueueMinusFlagOnPatch(), BlockPullRequest()])
+            results_summary = f'Found {num_issues} new failure{pluralSuffix} in {failing_files}'
+        elif num_issues:
+            pluralSuffix = 's' if num_issues > 1 else ''
+            # FIXME: Display which files are being ignored
+            results_summary = f'Ignored {num_issues} pre-existing failure{pluralSuffix}'
+        elif num_passes:
+            # FIXME: Add link to unexpected passes file
+            pluralSuffix = 's' if num_passes > 1 else ''
+            comment = f'Smart Pointer Build {formatted_build_link}: Found {num_passes} fixed file{pluralSuffix}!\n'
+            comment += 'Please update expectations using `smart-pointer-tool --update-expectations` before landing.'
+            self.setProperty('comment_text', comment)
+            self.build.addStepsAfterCurrentStep([LeaveComment()])
+            results_summary = f'Found {num_passes} fixed files: {passing_files}'
+
+        self.setProperty('build_finish_summary', results_summary)
+        if num_passes and num_failures:
+            results_summary += f' and found {num_passes} fixed files: {passing_files}'
+
+        return {u'step': results_summary}
+
+
+class ArchiveStaticAnalyzerResults(shell.ShellCommandNewStyle):
+    command = ["Tools/Scripts/generate-static-analysis-archive", "--id-string", WithProperties("Build #%(buildnumber)s"), "--output-root", SCAN_BUILD_OUTPUT_DIR, "--destination", "/tmp/static-analysis.zip"]
+    name = "archive-static-analyzer-results"
+    description = ["archiving static analyzer results"]
+    descriptionDone = ["archived static analyzer results"]
+    haltOnFailure = True
+
+
+class UploadStaticAnalyzerResults(UploadTestResults):
+    name = "upload-static-analyzer-results"
+    workersrc = "/tmp/static-analysis.zip"
```

Comment:
Better to store this path as a global variable and access that in multiple classes. Currently the risk is that someone might change it (in future) in one place and forget to update in another.